packages/legacy-client/src/ontology-runtime/ontologyProvider/calls/ClientContext.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { OntologyDefinition, ThinClient } from "@osdk/api";
+import type { OntologyObjectV2 } from "@osdk/gateway/types";
+import type { OntologyObject } from "../..";
+
+export interface ClientContext {
+  client: ThinClient<any>;
+  ontology: OntologyDefinition<any>;
+  createObject<T extends OntologyObject>(
```

Comment:
I think with the definition, we don't need createObject (in the existing world it was used to map back to the constructors from the generated code)

Reply:
Perhaps, I'm trying to figure out if there's benefits to passing this around or just relying on `convertWireToOsdkObject` as a global singleton.

Reply:
We can go with this for now and always change it if we want